webrtc/videoroom/lib/videoroom/display_manager.ex

```diff
@@ -0,0 +1,170 @@
+defmodule VideoRoom.DisplayManager do
+  @moduledoc false
+  # Module that manages endpoint's tracks i.e. it indicates which tracks should be sent and which disabled in order not
+  # to exceed maximal number of video tracks to display.
+  require Membrane.Logger
+
+  @type endpoint_id_t :: any()
+  @opaque t :: %__MODULE__{
+            displayed: %{},
+            queued: MapSet.t(),
+            rest: MapSet.t(),
+            max_display_num: non_neg_integer()
+          }
+
+  @enforce_keys [:displayed, :queued, :rest, :max_display_num]
+  defstruct @enforce_keys
+
+  def new(max_display_num) do
+    %__MODULE__{
+      displayed: %{},
+      queued: MapSet.new(),
+      rest: MapSet.new(),
+      max_display_num: max_display_num
+    }
+  end
+
+  @doc """
+  Adds new endpoint.
+
+  If some endpoint is going to send video tracks of endpoint with id `endpoint_id` then this function should be called.
+  """
+  @spec add(t(), endpoint_id :: endpoint_id_t()) :: t()
+  def add(state, endpoint_id) do
+    if map_size(state.displayed) < state.max_display_num,
+      do: put_in(state.displayed[endpoint_id], :silence),
+      else: %{state | rest: MapSet.put(state.rest, endpoint_id)}
+  end
+
+  @doc """
+  Updates voice activity status of endpoint with id `endpoint_id`.
+
+  Returns `{:replace, old_endpoint_id, new_endpoint_id}` if video track of `new_endpoint_id` should be replaced by
+  video track with `old_endpoint_id`.
+  Otherwise it returns `:ok`.
+  """
+  @spec update(t(), endpoint_id :: endpoint_id_t(), activity :: :speech | :silence) ::
+          {:ok
+           | {:replace, old_endpoint_id :: endpoint_id_t(), new_endpoint_id :: endpoint_id_t()},
+           t()}
+  def update(state, endpoint_id, :speech) do
+    cond do
+      Map.has_key?(state.displayed, endpoint_id) ->
+        state = put_in(state.displayed[endpoint_id], :speech)
+        {:ok, state}
+
+      MapSet.member?(state.queued, endpoint_id) ->
+        {:ok, state}
+
+      MapSet.member?(state.rest, endpoint_id) ->
+        case find_inactive(state) do
+          {nil, nil} ->
+            state = %{state | queued: MapSet.put(state.queued, endpoint_id)}
+            state = %{state | rest: MapSet.delete(state.rest, endpoint_id)}
+            {:ok, state}
+
+          {inactive_endpoint_id, :silence} ->
+            state = %{state | displayed: Map.delete(state.displayed, inactive_endpoint_id)}
+            state = %{state | rest: MapSet.put(state.rest, inactive_endpoint_id)}
+            state = %{state | rest: MapSet.delete(state.rest, endpoint_id)}
+            state = put_in(state.displayed[endpoint_id], :speech)
```

Comment:
I think a `move` utility function, that would enable moving endpoint id from one set to another, would make this prettier